File: io/github/zhengyhn/leetcode/pascals_triangle/pascals-triangle.cpp
/*
 * =====================================================================================
 *
 *       Filename:  pascals-triangle.cpp
 *
 *    Description:  打印帕斯卡三角
 *
 *        Version:  1.0
 *        Created:  08/02/2015 04:44:11 PM
 *       Revision:  none
 *       Compiler:  gcc
 *
 *         Author:  YOUR NAME (), 
 *   Organization:  
 *
 * =====================================================================================
 */

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector< vector<int> > generate(int numRows) {
      vector< vector<int> > result;

      for (int i = 0; i < numRows; ++i) {
        vector<int> row;
        row.push_back(1); 

        for (int j = 1; j < i; ++j) {
          row.push_back(result[i - 1][j - 1] + result[i - 1][j]);
        }
        if (i > 0 ) {
          row.push_back(1);
        }
        result.push_back(row);
      }
      return result;
    }
};

int main() {
  Solution sln;
  vector< vector<int> > result = sln.generate(5);
  
  for (int i = 0; i < result.size(); ++i) {
    for (int j = 0; j < result[i].size(); ++j) {
      cout << result[i][j] << " ";
    }
    cout << endl;
  }
}

File: io/github/zhengyhn/leetcode/pascals_triangle/pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  var result = [[]];    

  if (numRows <= 0) {
    return [];
  }

  for (var i = 0; i < numRows; ++i) {
    var row = [];
    row[0] = row[i] = 1;

    for (var j = 1; j < i; ++j) {
      row[j] = result[i - 1][j - 1] + result[i - 1][j];
    }
    result[i] = row;
  }

  return result;
};

console.log(generate(10));
